sklearn/compose/_column_transformer.py

```diff
@@ -672,6 +673,38 @@ def _validate_output(self, result):
                     "The output of the '{0}' transformer should be 2D (scipy "
                     "matrix, array, or pandas DataFrame).".format(name)
                 )
+        if _get_output_config("transform", self)["dense"] == "pandas":
+            return
+        try:
+            import pandas as pd
+        except ImportError:
+            return
```

Comment:
The fact that this line is not covered by our CI seems unexpected to me. I think it should be covered by CI configs where pandas is not installed and tests where we do not set change the default `set_output` config for "transform".

Reply:
when I run the tests locally in an environment without pandas this line is reached many times

Reply:
Here is where each of those lock files is used if it helps:

`build_tools/cirrus/py39_conda_forge_linux-aarch64_conda.lock`:
https://github.com/scikit-learn/scikit-learn/blob/ae2c80a21007d8295fb3ec3a927fce17b3d60ab8/build_tools/cirrus/arm_tests.yml#L11

`build_tools/azure/pypy3_linux-64_conda.lock`:
https://github.com/scikit-learn/scikit-learn/blob/ae2c80a21007d8295fb3ec3a927fce17b3d60ab8/azure-pipelines.yml#L122

`build_tools/azure/py38_conda_forge_mkl_win-64_conda.lock`:
https://github.com/scikit-learn/scikit-learn/blob/ae2c80a21007d8295fb3ec3a927fce17b3d60ab8/azure-pipelines.yml#L316

`build_tools/azure/pylatest_pip_scipy_dev_linux-64_conda.lock`:
https://github.com/scikit-learn/scikit-learn/blob/ae2c80a21007d8295fb3ec3a927fce17b3d60ab8/azure-pipelines.yml#L61

Reply:
if I don't catch the error, many tests fail [azure](https://dev.azure.com/scikit-learn/scikit-learn/_build/results?buildId=61422&view=logs&j=689a1c8f-ff4e-5689-1a1a-6fa551ae9eba&t=0b7e60d2-0e3c-59af-8129-1150b3e7bf0c&l=22306) so the line is indeed covered by the tests but not in builds with coverage enabled apparently